sql/planbuilder/aggregates.go

```diff
@@ -812,67 +813,55 @@ func (b *Builder) buildHaving(fromScope, projScope, outScope *scope, having *ast
 	if fromScope.groupBy == nil {
 		fromScope.initGroupBy()
 	}
-	// Having specifies conditions on groups. If not group by is present, all rows implicitly form a single aggregate group.
-	havingScope := fromScope.push() // TODO: we should not be including the entire fromScope
 
-	for _, c := range projScope.cols {
-		// if a projection overrides a column used in an aggregation, don't use projection
-		found := false
-		if fromScope.groupBy != nil && fromScope.groupBy.hasAggs() {
-			for _, cc := range fromScope.groupBy.aggregations() {
-				transform.InspectExpr(cc.scalar, func(e sql.Expression) bool {
-					switch e := e.(type) {
-					case *expression.GetField:
-						if strings.EqualFold(c.col, e.Name()) {
-							found = true
+	havingScope := b.newScope()
```

Comment:
could you add a docstring summarizing top-level rules? The before/after looks mostly the same just in a different order